test/contracts/oft/v2/NativeOFTV2.test.js

```diff
@@ -0,0 +1,454 @@
+
+        let leftOverAmount = ethers.utils.parseUnits("0", 18)
+        let totalAmount = ethers.utils.parseUnits("8", 18)
+
+        const ownerAddressBytes32 = ethers.utils.defaultAbiCoder.encode(["address"], [owner.address])
+        // estimate nativeFees
+        let nativeFee = (await nativeOFTV2.estimateSendFee(remoteChainId, ownerAddressBytes32, totalAmount, false, "0x")).nativeFee
+        expect(await ethers.provider.getBalance(localEndpoint.address)).to.be.equal(ethers.utils.parseUnits("0", 18))
+        await nativeOFTV2.sendFrom(
+            owner.address,
+            remoteChainId, // destination chainId
+            ownerAddressBytes32, // destination address to send tokens to
+            totalAmount, // quantity of tokens to send (in units of wei)
+            [owner.address, ethers.constants.AddressZero, "0x"],
+            { value: nativeFee.add(totalAmount.sub(depositAmount)) } // pass a msg.value to pay the LayerZero message fee
+        )
+
+        // expect(await ethers.provider.getBalance(owner.address)).to.be.equal(ownerBalance.sub(messageFee).sub(transFee).sub(depositAmount))
+        expect(await ethers.provider.getBalance(nativeOFTV2.address)).to.be.equal(totalAmount)
+        expect(await ethers.provider.getBalance(localEndpoint.address)).to.be.equal(nativeFee) // collects
+        expect(await nativeOFTV2.balanceOf(nativeOFTV2.address)).to.be.equal(totalAmount)
+        expect(await nativeOFTV2.balanceOf(owner.address)).to.be.equal(leftOverAmount)
+        expect(await remoteOFTV2.balanceOf(owner.address)).to.be.equal(totalAmount)
+
+        let ownerBalance2 = await ethers.provider.getBalance(owner.address)
+
+        const aliceAddressBytes32 = ethers.utils.defaultAbiCoder.encode(["address"], [alice.address])
+        // estimate nativeFees
+        nativeFee = (await nativeOFTV2.estimateSendFee(localChainId, aliceAddressBytes32, totalAmount, false, "0x")).nativeFee
+        await remoteOFTV2.sendFrom(
+            owner.address,
+            localChainId, // destination chainId
+            aliceAddressBytes32, // destination address to send tokens to
+            totalAmount, // quantity of tokens to send (in units of wei)
+            [owner.address, ethers.constants.AddressZero, "0x"],
+            { value: nativeFee.add(totalAmount) } // pass a msg.value to pay the LayerZero message fee
+        )
+
+        let transFee = ownerBalance2.sub(await ethers.provider.getBalance(owner.address)).sub(nativeFee)
+        expect(await ethers.provider.getBalance(alice.address)).to.be.equal(aliceBalance.add(totalAmount))
+        expect(await ethers.provider.getBalance(owner.address)).to.be.equal(ownerBalance2.sub(nativeFee).sub(transFee))
+        expect(await nativeOFTV2.balanceOf(owner.address)).to.equal(leftOverAmount)
+        expect(await remoteOFTV2.balanceOf(owner.address)).to.equal(0)
+    })
+
+    it("sendFrom() - with enough native", async function () {
```

Comment:
What is the difference between this test case and the first one?

Reply:
It hits different logic in _debitMsgSender